src/tests/functional/shared_test_classes/include/shared_test_classes/base/benchmark.hpp

```diff
@@ -0,0 +1,193 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <cctype>
+#include <iostream>
+
+#include "layer_test_utils.hpp"
+#include "ov_subgraph.hpp"
+
+namespace LayerTestsDefinitions {
+
+template <typename BaseLayerTest>
+class BenchmarkLayerTest : public BaseLayerTest {
+    static_assert(std::is_base_of<LayerTestsUtils::LayerTestsCommon, BaseLayerTest>::value,
+                  "BaseLayerTest should inherit from LayerTestsUtils::LayerTestsCommon");
+
+public:
+    static constexpr int kDefaultNumberOfAttempts = 100;
+
+    void RunBenchmark(const std::initializer_list<std::string>& node_type_names,
+                      const std::chrono::milliseconds warmupTime = std::chrono::milliseconds(2000),
+                      const int numAttempts = kDefaultNumberOfAttempts) {
+        bench_node_type_names_ = node_type_names;
+        warmup_time_ = warmupTime;
+        num_attempts_ = numAttempts;
+        LayerTestsUtils::LayerTestsCommon::configuration = {{"PERF_COUNT", "YES"}};
+        LayerTestsUtils::LayerTestsCommon::Run();
+    }
+
+    void RunBenchmark(const std::string& name,
+                      const std::chrono::milliseconds warmupTime = std::chrono::milliseconds(2000),
+                      const int numAttempts = kDefaultNumberOfAttempts) {
+        if (!name.empty()) {
+            RunBenchmark({name}, warmupTime, numAttempts);
+        } else {
+            RunBenchmark({}, warmupTime, numAttempts);
+        }
+    }
+
+    // NOTE: Validation is ignored because we are interested in benchmarks results.
+    //       In future the validate method could check if new benchmark results are not worse than previous one
+    //       (regression test), and in case of any performance issue report it in PR.
+    void Validate() override {
+    }
+
+protected:
+    void Infer() override {
+        this->inferRequest = this->executableNetwork.CreateInferRequest();
+        this->ConfigureInferRequest();
+
+        std::map<std::string, long long> results_us{};
+        for (const auto& node_type_name : bench_node_type_names_) {
+            results_us[node_type_name] = {};
+        }
+
+        // Warmup
+        auto warm_current = std::chrono::steady_clock::now();
+        const auto warm_end = warm_current + warmup_time_;
+        while (warm_current < warm_end) {
+            this->inferRequest.Infer();
+            warm_current = std::chrono::steady_clock::now();
+        }
+
+        // Benchmark
+        for (int i = 0; i < num_attempts_; ++i) {
+            this->inferRequest.Infer();
+            const auto& perf_results = LayerTestsUtils::LayerTestsCommon::inferRequest.GetPerformanceCounts();
```

Comment:
use the same code style please everywhere:
`this` or `LayerTestsUtils::LayerTestsCommon::` or call it directly (without the above constructions)
It's not so clear

Reply:
Done